src/main/java/net/minecraftforge/datafix/ForgeDataFixerCustomFixersModStateTransition.java

```diff
@@ -0,0 +1,220 @@
+
+            final List<Function<Schema, DataFix>> fixerFactories = internalEventGenerator.eventPerContainerMap.values()
+              .stream()
+              .flatMap(RegisterFixesEvent::getFixerFactories)
+              .toList();
+
+            internalEventGenerator.schema.registerCustomFixes(
+              fixerFactories
+            );
+        }, executor).handle((v, t) -> t != null ? Collections.singletonList(t) : Collections.emptyList());
+    }
+
+    /**
+     * Invoked by the event management system to indicate that it completely fired all events from the event stream.
+     * This indicates that all the events for all the schemas have been fired and post processed.
+     * As such we can now safely reconfigure the DFU wrapper so that the new schema configurations are picked
+     * up by the system.
+     *
+     * @param executor The executor to execute the task on.
+     * @param listCompletableFuture The list of previous failures.
+     * @return The future which indicates success or failure.
+     */
+    private static CompletableFuture<List<Throwable>> rebuildRulesAfterSchemaEvents(
+      final Executor executor, final CompletableFuture<List<Throwable>> listCompletableFuture
+    ) {
+        return listCompletableFuture.whenCompleteAsync((errors, except) -> {
+            //Check for errors.
+            if (except != null)
+            {
+                //Ouch that failed, log it and escape.
+                LOGGER.error("Detected errors during dfu setup. DFU might not be available.", except);
+            }
+            else
+            {
+                //Success, check for a compatible DFU instance and then rebuild it.
+                if (DataFixers.getDataFixer() instanceof ForgeDataFixerDelegateHandler forgeDataFixerDelegateHandler)
+                {
+                    //Do not do this on the mod executor!
+                    //This will crash the game.
+                    //Rebuild the DFU wrapper on the bootstrap executor
+                    forgeDataFixerDelegateHandler.rebuildFixer(Util.bootstrapExecutor(), true);
+                }
+            }
+        }, executor);
+    }
+
+    private static class InternalEventGenerator implements IModStateTransition.EventGenerator<RegisterFixesEvent> {
+
+        private static final Logger LOGGER = LogUtils.getLogger();
+
+        private final ForgeSchema                           schema;
+        private final Map<ModContainer, RegisterFixesEvent> eventPerContainerMap = Maps.newConcurrentMap();
+
+        private InternalEventGenerator(final ForgeSchema schema) {this.schema = schema;}
+
+        @Override
+        public RegisterFixesEvent apply(final ModContainer modContainer) {
+            return eventPerContainerMap.computeIfAbsent(modContainer, this::createEvent);
+        }
+
+        private RegisterFixesEvent createEvent(final ModContainer modContainer) {
```

Comment:
Formatting: braces on newlines.
```suggestion
        public RegisterFixesEvent apply(final ModContainer modContainer)
        {
            return eventPerContainerMap.computeIfAbsent(modContainer, this::createEvent);
        }

        private RegisterFixesEvent createEvent(final ModContainer modContainer)
        {
```